.github/workflows/ci.yml

```diff
@@ -79,11 +80,27 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all commits of this PR except HEAD compile
+        run: |
+          git reset --hard HEAD~1
```

Comment:
Can you add a comment here to mention why HEAD is skipped? Because it's compiled in other jobs.

Reply:
I tried both `HEAD~1` and `HEAD^1` but it seems to be pointing to before the PR begins. I'll leave the repo utouched for now

Reply:
the `HEAD` is probably the merge commit between the target branch and the PR's head

either of `HEAD^1` and `HEAD^2` should be the top level commit of the PR.
i don't know whether github guarantees  which is which